csrc/include/natten/cuda/fna/kernel_forward.h

```diff
@@ -288,24 +289,34 @@ struct FusedNeighborhoodAttentionKernel {
         output_accum_ptr = (accum_t*)output_ptr;
       }
 
-      if constexpr (kStoresLSE) {
-        if (logsumexp_ptr != nullptr) {
-          auto lse_stride_dilation = compute_stride(num_queries, num_heads);
-          lse_strideM = lse_stride_dilation * dilation;
-
-          // NOTE(alih): we don't pad for 128-bit alignment like in xFormers,
-          // because we have to rewrite the vector iterator in the backward
-          // kernel into a gather op.
-          // auto lse_dim = cutlass::ceil_div((int32_t)num_queries.prod32(),
-          // kAlignLSE) * kAlignLSE;
-          // lse[batch_id, query_start, head_id]
-          logsumexp_ptr += batch_id * num_queries.prod32() * num_heads +
-              ((first_query * lse_strideM).sum() +
-               (dilation_idx * lse_stride_dilation).sum()) +
+      if (logsumexp_ptr != nullptr) {
+        auto lse_stride_dilation = compute_stride(num_queries, num_heads);
+        lse_strideM = lse_stride_dilation * dilation;
+
+        // NOTE(alih): we don't pad for 128-bit alignment like in xFormers,
+        // because we have to rewrite the vector iterator in the backward
+        // kernel into a gather op.
+        // auto lse_dim = cutlass::ceil_div((int32_t)num_queries.prod32(),
+        // kAlignLSE) * kAlignLSE;
+        // lse[batch_id, query_start, head_id]
+        logsumexp_ptr += batch_id * num_queries.prod32() * num_heads +
+            ((first_query * lse_strideM).sum() +
+             (dilation_idx * lse_stride_dilation).sum()) +
+            head_id;
+      }
+      
+      if (maximums_ptr != nullptr){
+          // Following same pattern as lse strides
+          auto maximums_stride_dilation = compute_stride(num_queries, num_heads);
+          maximums_strideM = maximums_stride_dilation * dilation;
+          maximums_ptr += batch_id * num_queries.prod32() * num_heads + 
+              ((first_query * maximums_strideM).sum() +
+               (dilation_idx * maximums_stride_dilation).sum()) +
               head_id;
-        }
       }
```

Comment:
Can we simplify this a bit and compute the offset once separately, and add it to `logsumexp_ptr`, and then if `maximums_ptr != nullptr`, add exactly the same offset to the pointer. We shouldn't need to redo all of this.